beacon-chain/state/types.go

```diff
@@ -141,8 +143,11 @@ func (b *BeaconState) Copy() *BeaconState {
 
 	// Finalizer runs when dst is being destroyed in garbage collection.
 	runtime.SetFinalizer(dst, func(b *BeaconState) {
-		for _, v := range b.sharedFieldReferences {
+		for i, v := range b.sharedFieldReferences {
 			v.refs--
+			if i == randaoMixes && v.refs == 0 {
+				memorypool.PutDoubleByteSlice(b.state.RandaoMixes)
```

Comment:
Seems like there are many fields that could use this, like block roots, state roots, etc. want to include those too?

Reply:
I initially did the same for state roots/block roots. But in runtime i got a lot of wierd errors/corrupted db. So i just focused on optimizing the randao mixes